plugin/pkg/admission/webhook/config.go

```diff
@@ -0,0 +1,126 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package webhook checks a webhook for configured operation admission
+package webhook
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/apiserver/pkg/admission"
+)
+
+const (
+	defaultDuration           = time.Duration(0)
+	defaultFailAction         = Deny
+	defaultRetryBackoff       = time.Duration(500) * time.Millisecond
+	errInvalidFailAction      = "webhook rule (%d) has an invalid fail action: %s should be either %v or %v"
+	errInvalidRuleOperation   = "webhook rule (%d) has an invalid operation (%d): %s"
+	errInvalidRuleType        = "webhook rule (%d) has an invalid type: %s should be either %v or %v"
+	errMissingKubeConfigFile  = "kubeConfigFile is required"
+	errOneRuleRequired        = "webhook requires at least one rule defined"
+	errRetryBackoffOutOfRange = "webhook retry backoff is invalid: %v should be between %v and %v"
+	minRetryBackoff           = time.Duration(1) * time.Millisecond
+	maxRetryBackoff           = time.Duration(5) * time.Minute
+)
+
+// DefaultRule is a rule that will skip processing by default
+var (
+	DefaultRule = Rule{Type: Skip}
+)
+
+func normalizeConfig(config *GenericAdmissionWebhookConfig) error {
+	allowFailAction := string(Allow)
+	denyFailAction := string(Deny)
+	connectOperation := string(admission.Connect)
+	createOperation := string(admission.Create)
+	deleteOperation := string(admission.Delete)
+	updateOperation := string(admission.Update)
+	sendRuleType := string(Send)
+	skipRuleType := string(Skip)
+
+	// Validate the kubeConfigFile property is present
+	if config.KubeConfigFile == "" {
+		return fmt.Errorf(errMissingKubeConfigFile)
+	}
+
+	// Normalize and validate the retry backoff
+	if config.RetryBackoff == defaultDuration {
+		config.RetryBackoff = defaultRetryBackoff
+	} else {
+		// Unmarshalling gives nanoseconds so convert to milliseconds
+		config.RetryBackoff *= time.Millisecond
+	}
+
+	if config.RetryBackoff < minRetryBackoff || config.RetryBackoff > maxRetryBackoff {
+		return fmt.Errorf(errRetryBackoffOutOfRange, config.RetryBackoff, minRetryBackoff, maxRetryBackoff)
+	}
+
+	// Validate that there is at least one rule
+	if len(config.Rules) == 0 {
+		return fmt.Errorf(errOneRuleRequired)
+	}
+
+	for i, rule := range config.Rules {
+		// Normalize and validate the fail action
+		failAction := strings.ToUpper(string(rule.FailAction))
+
+		switch failAction {
+		case "":
+			config.Rules[i].FailAction = defaultFailAction
+		case allowFailAction:
+			config.Rules[i].FailAction = Allow
+		case denyFailAction:
+			config.Rules[i].FailAction = Deny
+		default:
+			return fmt.Errorf(errInvalidFailAction, i, rule.FailAction, Allow, Deny)
+		}
+
+		// Normalize and validate the rule operation(s)
+		for j, operation := range rule.Operations {
+			operation := strings.ToUpper(string(operation))
+
+			switch operation {
+			case connectOperation:
```

Comment:
The only way to make this work would be to use `admission.Operation(operation)` in your example above since `operation` is a `string` while we need an `admission.Operation`.  Since `admission.[Connect|Create|Delete|Update]` are constants, wouldn't this mean I'm creating new instances of `admission.Operation` instead of using one of those constants?